File: solutions/beecrowd/1031/1031.py
import sys


def josephus(n, k):
    if n == 1:
        return 1
    return ((josephus(n - 1, k) + k - 1) % n) + 1


def main():
    for line in sys.stdin:
        n = int(line)

        if not n:
            continue

        m = 1

        while josephus(n - 1, m) != 12:
            m += 1

        print(m)


main()

File: solutions/beecrowd/1031/1031.clj
(ns main)

(defn josephus [n k]
  (loop [current-n 2
         result 1]
    (if
     (> current-n n)
      result
      (recur (inc current-n)
             (-> result
                 (+ k)
                 dec
                 (mod current-n)
                 inc)))))

(defn calculate [n]
  (loop [m 1]
    (if (= 12 (josephus (dec n) m))
      m
      (recur (inc m)))))

(defn main []
  (loop [line (Integer/parseInt (read-line))]
    (printf "%d%n" (calculate line))
    (let [line' (Integer/parseInt (read-line))]
      (when (not= line' 0)
        (recur line')))))

(main)
